### DISCUSSION TITLE
Can not static link RmlUi 5

### DISCUSSION THREAD

**Discussion by xland** [mikke89/RmlUi#393] [Q&A] [answered]
I disabled the `BUILD_SHARED_LIBS` option.
![image](https://user-images.githubusercontent.com/2980915/207533360-9af28261-e63e-4623-b275-a236510201f3.png)

And then Visual Studio has static lib setting.
![image](https://user-images.githubusercontent.com/2980915/207533036-83ae75a3-3b4c-4753-a9ba-2f0709201ba7.png)

But when I build `ALL_BUILD` project,  I get the shared libararys.
![image](https://user-images.githubusercontent.com/2980915/207534410-96b7fffa-9c54-468a-bea0-4b3e43b4b060.png)

It won't happen on RmlUi 4.4
![image](https://user-images.githubusercontent.com/2980915/207535868-4da8de47-e23d-4846-abcd-6688885d2ba5.png)

Does this matter with GLFW+OpenGL3 backends

**Comment by mikke89** [comment]
I think you are in fact still building static libraries here. You can see from the time stamp on the `RmlCore.dll` that this file was built some hours earlier. So in all likelihood the dll file is just a remnant from a shared build you did earlier.

**Reply by xland** [reply]
But demo.exe dependent on RmlCore.dll.
demo project has `RMLUI_STATIC_LIB` preprocessor definition.

I rebuild demo project just now, and rename the RmlCore.dll.
(I think RmlCore.dll is cached by VisualStudio to avoid rebuild)
![image](https://user-images.githubusercontent.com/2980915/207992216-a7c85cc7-b0f1-45a9-baf8-b441ef2a8027.png)

**Comment by mikke89** [comment, marked as answer]
I can't replicate this, everything is working fine here on my end. `demo.exe` runs just fine.

![image](https://user-images.githubusercontent.com/5490330/207993812-2719b108-5d74-4cce-a88a-adb63e757966.png)

You already confirmed that MSVC is actually building static libraries, so the issue must be somewhere else. Perhaps the linker is linking in an earlier copy of RmlCore dynamic library from somewhere else? Do you perhaps use vcpkg, maybe that one links in another (dynamic) version of RmlUi. If so, try to disable vcpkg, or remove rmlui from it. Otherwise, check all the library paths for the `demo` project, and make sure that none of them have an old build of RmlUi.

**Reply by xland** [reply]
aha~~
Thanks very much.
It is caused by vcpkg.
